### DISCUSSION TITLE
[Support] DNS requests are not reaching Pi-Hole.

### DISCUSSION THREAD

**Discussion by starfishbzdf** [pivpn/pivpn#1486] [Support]
## In raising this issue, I confirm the following:

`{please fill the checkboxes, e.g: [X]}`

- [X] I have read the [documentation](https://docs.pivpn.io)
- [X] I have read and understood the [contributors guide](https://github.com/pivpn/pivpn/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*.
- [?] The issue I am reporting can be *is* directly related to the pivpn installer script.

### Have you searched for similar issues and solutions?
    Yes! These two posts seem to depict the same issue and they ran into a dead end:
    
    * https://redd.it/rd6dts
    * https://redd.it/q67yro
    
### Has your install failed? 
```
No!
```

#### What type of hardware are you running PiVPN at?

```
Raspberrypi Zero (with ethernet adapter)

```
#### Output of `uname -a`
```
 Linux raspberrypi 5.10.103+ #1529 Tue Mar 8 12:19:18 GMT 2022 armv6l GNU/Linux

```

#### Output of `cat /etc/os-release`
```
  PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

```

### If install failed Please provide the console output of  `curl -L https://install.pivpn.io | bash`
```
  
  
  
```

### Issue
I'm able to ping LAN clients, able to interact with Pi-Hole on both 192.168.1.4/admin and 10.6.0.1/admin, able to ping 8.8.8.8, but not able to ping google.com and google.com does not appear on the query log.

### Console output of      `curl -L install.pivpn.io | bash`
```

```

### Console output of      `pivpn add` or `pivpn add nopass`
```


```

### Console output of      `pivpn debug`

```
::: Generating Debug Output
::::            PiVPN debug              ::::
=============================================
::::            Latest commit            ::::
Branch: master
Commit: f80b0a7962d91862132c0a4abd65c1e67bd37bd7
Author: 4s3ti
Date: Fri Dec 3 21:41:13 2021 +0100
Summary: Update Changelog
=============================================
::::        Installation settings        ::::
PLAT=Raspbian
OSCN=buster
USING_UFW=0
IPv4dev=eth0
dhcpReserv=1
IPv4addr=192.168.1.4/24
IPv4gw=192.168.1.1
install_user=pi
install_home=/home/pi
VPN=wireguard
pivpnPORT=51820
pivpnDNS1=10.6.0.1
pivpnDNS2=
pivpnHOST=REDACTED
INPUT_CHAIN_EDITED=0
FORWARD_CHAIN_EDITED=0
pivpnPROTO=udp
pivpnMTU=1420
pivpnDEV=wg0
pivpnNET=10.6.0.0
subnetClass=24
ALLOWED_IPS="0.0.0.0/0, ::0/0"
UNATTUPG=1
INSTALLED_PACKAGES=(iptables-persistent wireguard-tools qrencode)
=============================================
::::  Server configuration shown below   ::::
[Interface]
PrivateKey = server_priv
Address = 10.6.0.1/24
MTU = 1420
ListenPort = 51820
### begin Note9 ###
[Peer]
PublicKey = Note9_pub
PresharedKey = Note9_psk
AllowedIPs = 10.6.0.2/32
### end Note9 ###
=============================================
::::  Client configuration shown below   ::::
[Interface]
PrivateKey = Note9_priv
Address = 10.6.0.2/24
DNS = 10.6.0.1

[Peer]
PublicKey = server_pub
PresharedKey = Note9_psk
Endpoint = REDACTED:51820
AllowedIPs = 0.0.0.0/0, ::0/0
=============================================
::::    Recursive list of files in       ::::
::::    /etc/wireguard shown below       ::::
/etc/wireguard:
configs
keys
wg0.conf

/etc/wireguard/configs:
clients.txt
Note9.conf

/etc/wireguard/keys:
Note9_priv
Note9_psk
Note9_pub
server_priv
server_pub
=============================================
::::            Self check               ::::
:: [OK] IP forwarding is enabled
:: [OK] Iptables MASQUERADE rule set
:: [OK] WireGuard is running
:: [OK] WireGuard is enabled (it will automatically start on reboot)
:: [OK] WireGuard is listening on port 51820/udp
=============================================
:::: Having trouble connecting? Take a look at the FAQ:
:::: https://docs.pivpn.io/faq
=============================================
:::: WARNING: This script should have automatically masked sensitive       ::::
:::: information, however, still make sure that PrivateKey, PublicKey      ::::
:::: and PresharedKey are masked before reporting an issue. An example key ::::
:::: that you should NOT see in this log looks like this:                  ::::
:::: YIAoJVsdIeyvXfGGDDadHh6AxsMRymZTnnzZoAb9cxRe                          ::::
=============================================
::::            Debug complete           ::::
:::
::: Debug output completed above.
::: Copy saved to /tmp/debug.log
:::

```
### Have you taken any steps towards solving your issue?
```
  I've looked for similar support seeking posts and tried the suggestions such as letting Pi-Hole listen on all interfaces from all sources, made sure it's accessible from the client and correctly configured as the DNS address, tested while disconnected from the home WiFI (cellular 4g). 
  Sought help in the matrix.org chat channel, was directed at here.
```

**Comment by coolapso** [comment]
thank you for reaching out and for filling everything adequately. 
Moving to discussions.

**Comment by orazioedoardo** [comment]
Assuming this

> letting Pi-Hole listen on all interfaces from all sources

still holds true (you don't need "from all sources" too though). Do you see requests from 10.6.0.X in the pihole log `tail -f /var/log/pihole.log`?

**Reply by starfishbzdf** [reply]
Sorry for the late reply. I ran the command and only see queries from 192.168.1.X (other clients active in the network), the sites I tried to access didn't even show up on the list.

**Reply by orazioedoardo** [reply]
Could you post the firewall configuration on the Pi `sudo iptables -vnL INPUT`? You can also use tcpdump to check whether requests are actually reaching the Pi: `sudo tcpdump -n -i wg0 port 53`. If they do, you should see some output.

**Reply by starfishbzdf** [reply]
```
pi@raspberrypi:~ $ sudo iptables -vnL INPUT
Chain INPUT (policy ACCEPT 6458K packets, 1470M bytes)
 pkts bytes target     prot opt in     out     source               destination 

```
Seems empty?

```
17:20:07.163558 IP 10.6.0.2.8740 > 10.6.0.1.53: 43698+ A? peoplestack-pa.googleapis.com. (47)
17:20:07.573967 IP 10.6.0.2.44331 > 10.6.0.1.53: 35600+ A? mtalk.google.com. (34)
17:20:08.631707 IP 10.6.0.2.34585 > 10.6.0.1.53: 15961+ AAAA? firebaseinstallations.googleapis.com. (54)
17:20:08.863634 IP 10.6.0.2.21334 > 10.6.0.1.53: 28543+ A? inbox.google.com. (34)
17:20:08.896585 IP 10.6.0.2.38716 > 10.6.0.1.53: 37834+ A? inbox.google.com. (34)
17:20:08.936713 IP 10.6.0.2.21382 > 10.6.0.1.53: 42717+ A? mail.google.com. (33)
17:20:09.171576 IP 10.6.0.2.8740 > 10.6.0.1.53: 43698+ A? peoplestack-pa.googleapis.com. (47)
17:20:09.671654 IP 10.6.0.2.7296 > 10.6.0.1.53: 3420+ A? dcg.microsoft.com. (35)
17:20:10.581562 IP 10.6.0.2.15959 > 10.6.0.1.53: 45594+ AAAA? mtalk.google.com. (34)
17:20:10.933591 IP 10.6.0.2.21382 > 10.6.0.1.53: 42717+ A? mail.google.com. (33)
17:20:11.651948 IP 10.6.0.2.45323 > 10.6.0.1.53: 7011+ A? firebaseinstallations.googleapis.com. (54)
17:20:11.652317 IP 10.6.0.2.7296 > 10.6.0.1.53: 3420+ A? dcg.microsoft.com. (35)
17:20:11.858657 IP 10.6.0.2.63801 > 10.6.0.1.53: 4782+ A? inbox.google.com. (34)
17:20:11.901610 IP 10.6.0.2.33785 > 10.6.0.1.53: 22960+ A? inbox.google.com. (34)
17:20:12.179567 IP 10.6.0.2.28027 > 10.6.0.1.53: 6398+ A? peoplestack-pa.googleapis.com. (47)
17:20:12.591580 IP 10.6.0.2.15959 > 10.6.0.1.53: 45594+ AAAA? mtalk.google.com. (34)
17:20:13.483653 IP 10.6.0.2.45564 > 10.6.0.1.53: 9824+ A? mail.google.com. (33)
17:20:13.631581 IP 10.6.0.2.45323 > 10.6.0.1.53: 7011+ A? firebaseinstallations.googleapis.com. (54)
17:20:13.873576 IP 10.6.0.2.63801 > 10.6.0.1.53: 4782+ A? inbox.google.com. (34)
17:20:13.901660 IP 10.6.0.2.33785 > 10.6.0.1.53: 22960+ A? inbox.google.com. (34)
17:20:13.931540 IP 10.6.0.2.17125 > 10.6.0.1.53: 52806+ A? mail.google.com. (33)
17:20:14.173551 IP 10.6.0.2.28027 > 10.6.0.1.53: 6398+ A? peoplestack-pa.googleapis.com. (47)
17:20:14.663480 IP 10.6.0.2.63006 > 10.6.0.1.53: 32250+ AAAA? dcg.microsoft.com. (35)
17:20:15.483657 IP 10.6.0.2.45564 > 10.6.0.1.53: 9824+ A? mail.google.com. (33)
17:20:15.563578 IP 10.6.0.2.7629 > 10.6.0.1.53: 65167+ A? mtalk.google.com. (34)
17:20:15.943621 IP 10.6.0.2.17125 > 10.6.0.1.53: 52806+ A? mail.google.com. (33)
17:20:16.651537 IP 10.6.0.2.63006 > 10.6.0.1.53: 32250+ AAAA? dcg.microsoft.com. (35)
17:20:16.881567 IP 10.6.0.2.13076 > 10.6.0.1.53: 16260+ A? inbox.google.com. (34)
17:20:17.571655 IP 10.6.0.2.7629 > 10.6.0.1.53: 65167+ A? mtalk.google.com. (34)
17:20:18.491614 IP 10.6.0.2.20370 > 10.6.0.1.53: 45020+ A? mail.google.com. (33)
17:20:18.901469 IP 10.6.0.2.13076 > 10.6.0.1.53: 16260+ A? inbox.google.com. (34)
17:20:19.491579 IP 10.6.0.2.41981 > 10.6.0.1.53: 58615+ A? mail.google.com. (33)
17:20:19.649553 IP 10.6.0.2.1357 > 10.6.0.1.53: 43533+ A? dcg.microsoft.com. (35)
17:20:20.473723 IP 10.6.0.2.20370 > 10.6.0.1.53: 45020+ A? mail.google.com. (33)
```
This is me trying to access the gmail app, the requests seem to appear but it still doesn't load.

**Comment by orazioedoardo** [comment]
> Seems empty?

No rules are blocking the traffic.

> This is me trying to access the gmail app, the requests seem to appear but it still doesn't load.

What's the listening behavior in the Pi-hole settings? https://docs.pi-hole.net/ftldns/interfaces/.

**Reply by starfishbzdf** [reply]
![image](https://user-images.githubusercontent.com/50381850/162773956-2df1f217-4b1e-474b-9e10-f0d5b3816604.png)

**Reply by orazioedoardo** [reply]
Can you run `pihole restartdns` and try again? I remember there was some race condition where wireguard started after pihole which made the resolver non responding on the wireguard interface.

**Reply by starfishbzdf** [reply]
I connected the client, then ran the command - didn't help. Should I be doing it in a different order?

**Reply by orazioedoardo** [reply]
No it's ok. Do you see any line like `Ignoring query from non-local network` in the pihole log? Was it before and/or after restarting?

**Reply by starfishbzdf** [reply]
> Ignoring query from non-local network

Negative, nothing like that. not in the pihole query log and not in the tcpdump lines. tried before and after restartdns.

**Reply by orazioedoardo** [reply]
I don't know what to try at this point, sorry. You may seek help on the pihole support forum. If you do find a solution, please report it here too.

**Comment by EagleIJoe** [comment]
What does the command _netstat -lnp | grep 53_ return?

**Reply by starfishbzdf** [reply]
tcp        0      0 0.0.0.0:53              0.0.0.0:*               LISTEN      501/pihole-FTL
tcp6       0      0 :::53                   :::*                    LISTEN      501/pihole-FTL
udp        0      0 0.0.0.0:53              0.0.0.0:*                           501/pihole-FTL
udp        0      0 0.0.0.0:36494           0.0.0.0:*                           253/avahi-daemon: r
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           253/avahi-daemon: r
udp6       0      0 :::53                   :::*                                501/pihole-FTL
udp6       0      0 :::52423                :::*                                253/avahi-daemon: r
udp6       0      0 :::5353                 :::*                                253/avahi-daemon: r
